File: binding_rv_adapter/src/main/java/me/lx/rv/itembindings/OnItemBindModel.java
package me.lx.rv.itembindings;

import androidx.annotation.NonNull;

import me.lx.rv.ItemBinding;
import me.lx.rv.OnItemBind;

/**
 * An {@link OnItemBind} that selects item views by delegating to each item. Items must implement
 * {@link ItemBindingModel}.
 */
public class OnItemBindModel<T extends ItemBindingModel> implements OnItemBind<T> {

    @Override
    public void onItemBind(@NonNull ItemBinding<?> itemBinding, int position, T item) {
        item.onItemBind(itemBinding);
    }
}

File: binding_rv_adapter/src/main/java/me/lx/rv/itembindings/OnItemBindModel.kt
package me.lx.rv.itembindings

import me.lx.rv.OnItemBind
import me.lx.rv.XmlItemBinding

/**
 * An [OnItemBind] that selects item views by delegating to each item. Items must implement
 * [ItemBindingModel].
 */
class OnItemBindModel<T : ItemBindingModel> : OnItemBind<T> {

    override fun onItemBind(itemBinding: XmlItemBinding<*>, position: Int, item: T) {
        item.onItemBind(itemBinding)
    }
}
